CHANGELOG.md

```diff
@@ -12,10 +12,12 @@
 - Updated `pkg/test/e2eutil.WaitForDeployment()` and `pkg/test/e2eutil.WaitForOperatorDeployment()` to successfully complete waiting when the available replica count is _at least_ (rather than exactly) the minimum replica count required. ([#2248](https://github.com/operator-framework/operator-sdk/pull/2248))
 - Replace in the Ansible based operators module tests `k8s_info` for `k8s_facts` which is deprecated. ([#2168](https://github.com/operator-framework/operator-sdk/issues/2168))
 - Upgrade the Ansible version from `2.8` to `2.9` on the Ansible based operators image. ([#2168](https://github.com/operator-framework/operator-sdk/issues/2168))
+- **Breaking change:** Renamed `operator-sdk generate openapi` to `operator-sdk generate crds`. ([#2276](https://github.com/operator-framework/operator-sdk/pull/2276))
```

Comment:
I am not sure if we should move forward with. See that kubebuilder is not supporting it and users are asking for [here](https://github.com/kubernetes-sigs/kubebuilder/issues/1129#issuecomment-558704012) and a track issue from them [here](https://github.com/kubernetes-sigs/kubebuilder/issues/1081) which in my understand shows that require it as well.

Reply:
The first issue is fair; based on feedback in #2042 some users do not use `zz_generated.openapi.go`, which, on top of it not being necessary for API code unlike `zz_generated.deepcopy.go`, is why I decided to push for removing it. We should either:

1. Add documentation on building and using [`openapi-gen`](https://github.com/kubernetes/kube-openapi/tree/master/cmd/openapi-gen) to generate the files yourself, which is fairly straightforward.
2. Keep `generate openapi` as a command but move CRD generation into `generate crd`.

I vote for option 1 since we can offer Go OpenAPI code by wrapping `openapi-gen` in a Makefile recipe (in the future).

The second issue describes something we do not currently support in CRDs anyway since we use the same tool as kubebuilder to generate CRDs, [`controller-gen`](https://github.com/kubernetes-sigs/controller-tools/blob/6eef398/cmd/controller-gen/main.go) from controller-tools. Generating `zz_generated.openapi.go` does not change the state of supported annotations for CRD manifest validation in the SDK.

We agreed to make this change because it de-couples CRD generation from OpenAPI spec generation, two different albeit related processes. This change will help users debug errors more easily as well.

Reply:
Another option would be to add `generate crds` but keep `generate openapi` as is and deprecate it. We could hide it from the help output and log a deprecation message when it is called.

Then in a later version it could be removed.

Reply:
And we could/should still do one of the options @estroz suggested.

Reply:
@joelanford I like the combination of what you suggested and my first option. The docs referencing Go OpenAPI spec generation (I believe the only documentation doing so is the one being changed in this PR already) should warn that the subcommand is deprecated and users should switch to installing and running `openapi-gen` themselves if they still want that spec file.

Reply:
I think the idea of phasing out the old command with a deprecation warning. 

@estroz @joelanford we need to make it easy to download and use the openapi-gen command if that is the guidance, we either do it for the user and attempt to install the binary into a known location and give them the option to supply their own openapi-gen. 

Could we vendor the openapi-gen command and use it in the operator-sdk command? was that an option already looked?

Reply:
@shawn-hurley we currently vendor the exposed `openapi-gen` generator in `generate openapi`. If we want to install this binary for the user, I suggest this as the first candidate for a Makefile:
```make
openapi-gen:
	which ./bin/openapi-gen > /dev/null || go build -o ./bin/openapi-gen k8s.io/kube-openapi/cmd/openapi-gen
	./bin/openapi-gen ...
```
Otherwise documentation to the effect of the above script should suffice. I'd rather have documentation than a make recipe.

Reply:
By the way, our vendoring of the code generation libraries is what causes the GOROOT and panic issues described in this issue: https://github.com/operator-framework/operator-sdk/issues/1854#issuecomment-525132306

If we direct users to build `openapi-gen` from source like Eric suggests, we'll help them avoid that issue, at least for the openapi generation (I think it will still be an issue for deepcopy generation).

Reply:
Alright, updated the PR to keep `generate openapi`, but to hide it from help and deprecate it.